retriever/lib/retrieval.js

```diff
@@ -18,27 +18,32 @@
 export async function retrieveFile(
   baseUrl,
   pieceCid,
+  request,
   cacheTtl = 86400,
   { signal } = {},
 ) {
   const url = getRetrievalUrl(baseUrl, pieceCid)
-  const response = await fetch(url, {
-    cf: {
-      cacheTtlByStatus: {
-        '200-299': cacheTtl,
-        404: 0,
-        '500-599': 0,
-      },
-      cacheEverything: true,
-    },
-    signal,
-  })
-  const cacheStatus = response.headers.get('CF-Cache-Status')
-  if (!cacheStatus) {
-    console.log(`CF-Cache-Status was not provided for ${url}`)
-  }
 
-  const cacheMiss = cacheStatus !== 'HIT'
+  const cacheKey = new Request(url, request)
+  let response = await caches.default.match(cacheKey)
```

Comment:
Did you try adding [@cloudflare/worker-types](https://www.npmjs.com/package/@cloudflare/workers-types?activeTab=readme) package? Maybe that could resolve this issue.